cfpb/consumerfinance.gov
# TCCP: Assorted user interface improvements

Handful of improvements that came out of user testing.

See https://github.local/Design-Development/Design-Thinking-and-User-Research/issues/296

## Changes

- Form validation for landing page "location" field
- Visual linear gradient over the 11th card result to indicate that there are more cards to view.
- New "see card details" link
- Speed bumps are larger
- Misc. hierarchy changes

## How to test this PR

1. `yarn cypress run --spec test/cypress/integration/consumer-tools/credit-cards/explore-cards.cy.js`
1. Visit the [landing page](http://localhost:8000/consumer-tools/credit-cards/explore-cards) and see what happens when you try to submit the form without selecting a location.
1. Visit the [cards page](http://localhost:8000/consumer-tools/credit-cards/explore-cards/cards/?situations=Transfer%20a%20balance&situations=Build%20credit&credit_tier=Credit%20scores%20from%20620%20to%20719&location=IN&ordering=transfer_apr) and change the "sort by" field. The results are capped at 10 and a "show more" button appears *except* when sorting by card name.

## Screenshots

| "show more" fade | location validation | "see card details" link | 
|--------|-------|-------|
|<img width="1145" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/1060248/f42a6fc2-7e54-4dad-86c3-275df2e77e4c">|<img width="518" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/1060248/af097c0c-dd36-4b2c-942b-cd400141f74a">|<img width="393" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/1060248/8462de15-e7d4-4677-a02e-8caf016edbdb">|

## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
